cluster.go

```diff
@@ -222,10 +217,18 @@ func (c *clusterClient) _refresh() (err error) {
 
 	groups := result.parse(c.opt.TLSConfig != nil)
 	conns := make(map[string]connrole, len(groups))
+	masterOpt := *c.opt
+	if c.opt.SendToReplicas != nil {
+		masterOpt.ReplicaOnly = false
+	}
+	replicaOpt := *c.opt
+	if c.opt.SendToReplicas != nil {
+		replicaOpt.ReplicaOnly = !c.opt.ReplicaOnly
+	}
```

Comment:
It is better now. But it still allocates `masterOpt` and `replicaOpt` in every refresh.
We can make it even better by storing a `c.replicaOpt` in the `newClusterClient`.

Reply:
>  i just want to send "READONLY" command to replica connections automatically.

Agreed. So, how about this:

when `SendToReplicas` is configured and `ReplicaOnly` is true,

* return `ErrReplicaOnlyConflict`

when SendToReplicas is configured and ReplicaOnly is false,

* connection to master: use `c.connFn(addr, c.opt)`
* connection to replica: use `c.connFn(addr, c.replicaOpt)`

when SendToReplicas is not configured and ReplicaOnly is true,

* connection to all nodes: use `c.connFn(addr, c.opt)`, we don't need `c.replicaOpt` in this case.

Reply:
[5702ce3](https://github.com/redis/rueidis/pull/430/commits/5702ce3e8deb82dd5848d1adba36b07842e94e57)

Good! i changed it.